mobly/controllers/android_device_lib/snippet_client_v2.py

```diff
@@ -276,24 +317,274 @@ def _read_protocol_line(self):
+
+    Raises:
+      UnicodeError: if failed to decode the given bytes using encoding utf8.
+    """
+    try:
+      return str(response, encoding='utf8')
+    except UnicodeError:
+      self.log.error(
+          'Failed to decode socket response bytes using encoding '
+          'utf8: %s', response)
+      raise
+
+  def handle_callback(self, callback_id, ret_value, rpc_func_name):
+    """Creates the callback handler object.
+
+    If the client doesn't have an event client, it will start an event client
+    before creating a callback handler.
+
+    Args:
+      callback_id: see base class.
+      ret_value: see base class.
+      rpc_func_name: see base class.
+
+    Returns:
+      The callback handler object.
+    """
+    if self._event_client is None:
+      self._create_event_client()
+    return callback_handler.CallbackHandler(callback_id=callback_id,
+                                            event_client=self._event_client,
+                                            ret_value=ret_value,
+                                            method_name=rpc_func_name,
+                                            ad=self._device)
+
+  def _create_event_client(self):
+    """Creates a separate client to the same session for propagating events.
+
+    As the server is already started by the snippet server on which this
+    function is called, the created event client connects to the same session
+    as the snippet server. It also reuses the same host port and device port.
+    """
+    self._event_client = SnippetClientV2(package=self.package, ad=self._device)
+    self._make_connection_for_event_client(self.uid,
+                                           ConnectionHandshakeCommand.CONTINUE)
+
+  def _make_connection_for_event_client(self,
+                                        uid=UNKNOWN_UID,
+                                        cmd=ConnectionHandshakeCommand.INIT):
+    """Makes a connection to the server for the event client.
+
+    The process of making a connection for the event client is slightly
+    different from the snippet client. Each event client must belong to one
+    snippet client, so the event client reuses the port forwarding process
+    of the snippet client. Thus instead of calling `_make_connection`, it calls
+    the methods to reset the RPC id counter, create a socket connection and send
+    a handshake request.
+
+    Args:
+      uid: int, the uid of the server session to continue. It will be ignored
+        if the `cmd` requires the server to create a new session.
+      cmd: ConnectionHandshakeCommand, the handshake command Enum for the
+        server, which requires the server to create a new session or use the
+        current session.
+    """
+    self._event_client.host_port = self.host_port
+    self._event_client.device_port = self.device_port
+    self._event_client._counter = self._event_client._id_counter()  # pylint: disable=protected-access
```

Comment:
there are many other access of private members, but only this one has `pylint` comment.
is this necessary?

Reply:
Yes, it's necessary. Because the `_counter` is a private member of the event client object, I think elsewhere we only access the private member of `self`. 

Also verified that gpylint will warn about it if delete this pylint comment.

Reply:
if `_id_counter` needs to be accessed, why not just make it "public"?

Reply:
Seems that `_counter` is better to be private as it is just the counter of RPC ID and users should not care about it. So refactored this function to avoid access to this private member.